cargo-pgrx/src/command/upgrade.rs

```diff
@@ -0,0 +1,235 @@
+                self.allow_prerelease,
+                None,
+                path,
+                Some(&reg_url),
+            )
+            .map_err(|e| {
+                eyre!(
+                    "Unable to fetch the latest version \
+                        for crate {dep_name} due to {e}"
+                )
+            })?
+            .version()
+            .map(|s| s.to_string()),
+        };
+        let target_version = match target_version {
+            Some(ver) => ver,
+            None => {
+                return Err(eyre!(
+                    "Unable to update {dep_name} \
+                , no provided crate version and a latest version \
+                could not be retrieved from the registry."
+                ))
+            }
+        };
+
+        // As of cargo-edit version 0.12.2, dep.source will always be a Some()
+        // if the Dependency struct was instantiated via from_toml().
+        let source = match parsed_dep.source().cloned() {
+            Some(src) => src,
+            None => {
+                return Err(eyre!(
+                    "Dependency {dep_name}'s source was \
+                parsed as None by cargo-edit."
+                ))
+            }
+        };
+        debug!(
+            "Found dependency {dep_name} with current \
+            source {source:#?}"
+        );
+        match &source {
+            cargo_edit::Source::Registry(_) => {
+                if get_dep_source_version(&source).is_some() {
+                    replace_version(
+                        target_version.as_str(),
+                        path.as_path(),
+                        &mut key,
+                        dep,
+                        parsed_dep,
+                        source,
+                    )?;
+                } else {
+                    info!("No version specified for {dep_name}, not upgrading.");
+                }
+            }
+            cargo_edit::Source::Path(_) => {
+                warn!(
+                    "Cannot upgrade the version of \
+                        {dep_name} because it is a path (local) dependency."
+                )
+            }
+            cargo_edit::Source::Git(_) => {
+                warn!(
+                    "Cannot upgrade the version of \
+                        {dep_name} because it is a git dependency."
+                )
+            }
+            cargo_edit::Source::Workspace(_) => {
+                warn!(
+                    "Cannot upgrade the version of \
+                        {dep_name} because it is set in the \
+                        workspace, please run `cargo pgrx \
+                        upgrade` in the workspace directory."
+                )
+            }
+        }
+        Ok(())
+    }
+}
+
+impl CommandExecute for Upgrade {
+    #[tracing::instrument(level = "error", skip(self))]
+    fn execute(self) -> eyre::Result<()> {
+        const RELEVANT_PACKAGES: [&str; 3] = ["pgrx", "pgrx-macros", "pgrx-tests"];
```

Comment:
I guess we don't need `pgrx-pg-sys` here since people should get it through `pgrx::pg_sys`. This list seems a little fragile, though.

Reply:
We should still bump pgrx-pg-sys if we find it.

Reply:
Same with pgrx-sql-entity-graph.

Reply:
These should be present in the latest revision, but it'd be good to double-check this.